codex/conf.nim

```diff
@@ -337,6 +346,18 @@ proc readValue*(r: var TomlReader, val: var SignedPeerRecord) =
 
   val = SignedPeerRecord.parseCmdArg(uri)
 
+proc readValue*(r: var TomlReader, val: var MultiAddress) =
+  without input =? r.readValue(string).catch, err:
+    error "invalid MultiAddress configuration value", error = err.msg
+    quit QuitFailure
+
+  let res = MultiAddress.init(input)
```

Comment:
Same, `without` is probably more straight forward here.

Reply:
I'm not sure if that works since `val` is an argument? Though maybe it'd work with a `val = res` at the end.